frontend/components/DocumentFilters/DocumentFilters.jsx

```diff
@@ -24,36 +24,61 @@ export default class DocumentFilters extends Component {
     /**
      * The JSON-stringified object of filters currently applied.
      */
-    filter: proptypes.string
+    filter: proptypes.string,
+
+    /**
+     * Whether the filters are visible.
+     */
+    visible: proptypes.bool
   }
 
   constructor(props) {
     super(props)
 
-    const {collection, filter} = props
+    const {collection} = this.props
+    const paramFilters = this.setFiltersFromParams()
+    this.state.filters = paramFilters.length ? paramFilters : this.createDefaultFilters(collection)
+  }
+
+  setFiltersFromParams () {
+    const {filters} = this.props
+    return (filters ? objectToArray(filters, 'field') : [])
+      .map(this.deconstructFilters)
+  }
 
-    // Evaluate passed filter / store in state
-    const filters = (isValidJSON(filter) ? objectToArray(JSON.parse(filter), 'field') : []).map(this.deconstructFilters)
+  componentWillUpdate(nextProps, nextState) {
+    const {collection, filters} = this.props
 
-    if (!filters.length) {
-      filters.push({
-        index: 0,
-        field: Object.keys(collection.fields)[0],
-        type: '$eq'
-      })
+    if (nextProps.collection !== collection) {
+      // If we're changing collection, reset all filters
+      this.setState({filters: this.createDefaultFilters(nextProps.collection)})
+    } else {
+      // If we aren't changing collection
+      const paramFilters = this.setFiltersFromParams()
+      this.setState({filters: paramFilters.length ? paramFilters : this.createDefaultFilters(collection)})
     }
+  }
+
+  createDefaultFilters(collection) {
+    const filters = [{
+      field: Object.keys(collection.fields)[0],
+      type: '$eq',
+      value: null
+    }]    
 
-    // The list of applied filters
-    this.state.filters = filters
+    return filters
   }
 
   render() {
     const {filters} = this.state
-    const {collection} = this.props
+    const {collection, visible} = this.props
+
+    if (!visible) return null
 
     return (
       <form class={styles.filters} onSubmit={e => e.preventDefault()}>
-        {filters && collection && filters.map((filter, index) => (
+        {filters && collection && filters.map((filter, index) => {
+          return (
           <DocumentFilter
```

Comment:
I meant to remove the return, it was there for testing so I could output filter to the console. Sorted now